src/erlang_ls_document_symbol_provider.erl

```diff
@@ -0,0 +1,47 @@
+-module(erlang_ls_document_symbol_provider).
+
+-behaviour(erlang_ls_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        , setup/1
+        , teardown/0
+        ]).
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% erlang_ls_provider functions
+%%==============================================================================
+
+-spec is_enabled() -> boolean().
+is_enabled() ->
+  true.
+
+-spec setup(map()) -> erlang_ls_provider:state().
+setup(_Config) ->
+  #{}.
+
+-spec handle_request(any(), erlang_ls_provider:state()) ->
+  {any(), erlang_ls_provider:state()}.
+handle_request({document_symbol, Params}, State) ->
+  #{ <<"textDocument">> := #{ <<"uri">> := Uri}} = Params,
+  {ok, Document} = erlang_ls_db:find(documents, Uri),
+  POIs = erlang_ls_document:points_of_interest(Document),
+  case POIs of
+    [] -> {null, State};
+    _ ->
+      {[ #{ name => list_to_binary(io_lib:format("~p/~p", [F, A]))
+          , kind => ?SYMBOLKIND_FUNCTION
+          , location => #{ uri   => Uri
+                         , range => erlang_ls_protocol:range(Range)
+                         }
+          }
+         || #{ info := {function, {F, A}}
```

Comment:
Is there any other kind of symbol we may want to return? Like `define`s for example, we could return them as `SYMBOLKIND_CONSTANT`. If you think it's worth it, we can create a separate issue and it can be left for another PR.

Reply:
Probably. For now I just wanted to replace the `imenu` usage I have with EDTS which only works with functions. We can add the rest separately.